examples/text-classification/run_glue.py

```diff
@@ -353,20 +391,29 @@ def compute_metrics(p: EvalPrediction):
         train_result = trainer.train(
             model_path=model_args.model_name_or_path if os.path.isdir(model_args.model_name_or_path) else None
         )
-        metrics = train_result.metrics
-
-        trainer.save_model()  # Saves the tokenizer too for easy upload
+        ##### $custom ####
+        if is_run_on_sagemaker():
+            trainer.save_model(os.environ["SM_MODEL_DIR"])  # Saves the tokenizer too for easy upload
+        else:
+            trainer.save_model()  # Saves the tokenizer too for easy upload
+        ####  $custom end ####
 
         output_train_file = os.path.join(training_args.output_dir, "train_results.txt")
         if trainer.is_world_process_zero():
             with open(output_train_file, "w") as writer:
                 logger.info("***** Train results *****")
-                for key, value in sorted(metrics.items()):
+                for key, value in sorted(train_result._asdict().items()):
```

Comment:
This is not the same as the metrics, I don't understand this change, could you explain why you're doing it?

Reply:
The same goes for this. Since I removed the `metrics` i worked with the TrainingOuput Tuple and converted it to a dict to beable to iterate it. But I am going to revert it too.